### PR TITLE
Add support for biometrics

### PR DESCRIPTION
**Pull request by vishnuravi**
# Add support for biometrics

## :recycle: Current situation & Problem
The module currently allows for using a code with access guards. Given that most iOS devices have either Face ID or Touch ID we would like to allow developers to enable these for access guards if available and fall back to a code if not. This is also mentioned in Issue #6.

## :gear: Release Notes 
Implements Touch ID and Face ID-based access guards with a code as a backup.

Other:
- Migrates to string catalogs.
- Updates documentation with more detailed instructions for setting up different types of access guards.

## :books: Documentation
Requires `NSFaceIDUsageDescription` to be added to `Info.plist` in the main target for FaceID. Documentation has been updated to note this.

## :white_check_mark: Testing
There does not appear to be a way to UI test Face ID.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziAccessGuard/Extensions/LAContext+Async.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import LocalAuthentication
+
+
+extension LAContext {
+    func evaluatePolicyAsync(_ policy: LAPolicy, localizedReason: String) async throws -> Bool {
```
Comment (PSchmiedmayer) [inline Sources/SpeziAccessGuard/Extensions/LAContext+Async.swift:22]: Nice extension! 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the additions, the chances look great! 🎉

I would only suggest to add a section to the README and DocC documentation about the potential requirement to add elements it the Info.plist in case the developer wants to use the biometric functionality. We do something similar in the README of Spezi Speech: https://github.com/StanfordSpezi/SpeziSpeech#3-configure-target-properties and the DocC documentation there: https://swiftpackageindex.com/stanfordspezi/spezispeech/documentation/spezispeechrecognizer#3-Configure-target-properties 👍
I would also be great to mention biometrics support in the README and the DocC landing page.

Apart from these suggestions I would be happy to see the PR merged; thanks for all the work here @vishnuravi! 🚀

**Comment by vishnuravi** [comment]
> Thank you for the additions, the chances look great! 🎉
> 
> I would only suggest to add a section to the README and DocC documentation about the potential requirement to add elements it the Info.plist in case the developer wants to use the biometric functionality. We do something similar in the README of Spezi Speech: https://github.com/StanfordSpezi/SpeziSpeech#3-configure-target-properties and the DocC documentation there: https://swiftpackageindex.com/stanfordspezi/spezispeech/documentation/spezispeechrecognizer#3-Configure-target-properties 👍 I would also be great to mention biometrics support in the README and the DocC landing page.
> 
> Apart from these suggestions I would be happy to see the PR merged; thanks for all the work here @vishnuravi! 🚀

Thank you for the suggestions - everything has been addressed!

**Review by PSchmiedmayer — APPROVED** [review]
Nice, thank you for all the changes; this looks great @vishnuravi! 🚀

**Pull request merged**